safetrace-app/Features/Common/WebViewController.swift

```diff
@@ -89,6 +130,95 @@ final class WebViewController: UIViewController {
     }
 }
 
+extension WebViewController: WKScriptMessageHandler {
+    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
+
+        guard let messageUrl = message.webView?.url, WebViewHelper.isAuthorizedDomain(url: messageUrl) else {
+            print("Unauthorized webview interface. Message body: \(message.body)")
+            return
+        }
+
+        guard let body = message.body as? String else {
+            print("Unexpected message body for \"\(message.name)\": \(message.body)")
+            return
+        }
+
+        switch message.name {
+        case "contactTracing":
+            if body == "getOptInStatus" {
+                let isOptedIn = environment.safeTrace.isOptedIn
+                let javascriptToRun: String = "_tracing.setOptedInStatus(\(isOptedIn))"
+                runJavaScript(javascriptToRun)
+            } else if body == "openOptInPage" {
+                presentContactTracingController()
+            } else if body == "getIsCitizenInstalled" {
+                let citizenURL = Constants.citizenDeeplinkUrl
+                let isCitizenInstalled = UIApplication.shared.canOpenURL(citizenURL)
+
+                let javascriptToRun: String = "_tracing.setIsCitizenInstalled(\(isCitizenInstalled))"
+                runJavaScript(javascriptToRun)
+            } else if body == "optInToTracing" {
+                environment.safeTrace.startTracing()
+            }
+        case "user":
+//            if body == "getLocation" {
```

Comment:
why is this here?

Reply:
I can remove this. Was initially assuming that location is coming right up

Reply:
ah, got it, no need to remove i guess